src/store/store.ts

```diff
@@ -2,53 +2,49 @@ import { IUserData } from '@api';
 import { mergeDeepLeft } from 'ramda';
 import { useEffect } from 'react';
 import { unstable_batchedUpdates } from 'react-dom';
-import create, { GetState, SetState, StateCreator, StoreApi } from 'zustand';
+import create, { GetState, Mutate, SetState, StoreApi } from 'zustand';
 import createContext from 'zustand/context';
-import { devtools, persist, StoreApiWithDevtools, StoreApiWithPersist } from 'zustand/middleware';
+import { devtools, NamedSet, persist } from 'zustand/middleware';
 import { docsSlice, searchSlice, themeSlice, userSlice } from './slices';
 import { AppState } from './types';
 
 export const APP_STORAGE_KEY = 'nectar-app-state';
 
-// simple state logger, override one of these to log out for debugging state
-const log =
-  (config: StateCreator<AppState>) => (set: SetState<AppState>, get: GetState<AppState>, api: StoreApi<AppState>) =>
-    config(set, get, api);
-
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 export const createStore = (preloadedState: Partial<AppState> = {}) => {
+  const state = (set: NamedSet<AppState>, get: GetState<AppState>) => ({
+    ...searchSlice(set, get),
+    ...docsSlice(set, get),
+    ...userSlice(set, get),
+    ...themeSlice(set, get),
+    ...preloadedState,
+  });
+
+  // return a basic store during testing
+  if (process.env.NODE_ENV === 'test') {
+    return create<AppState>(state);
+  }
+
   const store = create<
     AppState,
     SetState<AppState>,
     GetState<AppState>,
-    StoreApiWithPersist<AppState> & StoreApiWithDevtools<AppState>
+    Mutate<StoreApi<AppState>, [['zustand/persist', Partial<AppState>], ['zustand/devtools', never]]>
```

Comment:
@thostetler Can you help me understand what these are?

Reply:
Hey @shinyichen, yeah this was a suggested deprecation in the types when I was refactoring
I just looked at https://github.com/pmndrs/zustand/blob/0fce3a9e7460002cf66222c86138095a47474121/tests/middlewareTypes.test.tsx#L380

to find out how to handle these types.

This is basically so that the middlewares (`devtools` and `persist`) are properly typed.  This is a cleaner way of doing things, so much preferred!